components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/model/graph/JsBaseGraphBuilderFactory.java

```diff
@@ -23,14 +23,13 @@
 
 import java.util.Map;
 
-import javax.script.ScriptEngine;
-
 /**
  * Interface for Factory to create a Javascript based sequence builder.
  * This factory is there to reuse of script engine and any related expensive objects.
  *
+ * @param <T> Type of the Javascript Engine
  */
-public interface JsBaseGraphBuilderFactory {
+public interface JsBaseGraphBuilderFactory<T> {
```

Comment:
Check the possibility of introducing a new interface

Reply:
It is not possible to introduce a new interface with the type parameter and extend this interface from the new interface as `JsBaseGraphBuilderFactory extends JsCommonGraphBuilderFactory<ScriptEngine>`.
We would anyways need to change the public methods in framework service data holder to access this interface to support GralJS based implementation. 
Hence keeping this change.